### PR TITLE
fast

### PR DESCRIPTION
**Pull request by mhevery**
Make the benchmark run 10x faster

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import {bootstrap} from 'angular2/angular2';
+import {bootstrap, ChangeDetection, jitChangeDetection} from 'angular2/angular2';
+import {bind} from 'angular2/di';
 import {Calendar} from 'components/calendar/calendar';
 import {CalendarCell} from 'components/calendar-cell/calendar-cell';
 
 export function main() {
-  return bootstrap(Calendar);
+  return bootstrap(Calendar, [bind(ChangeDetection).toValue(jitChangeDetection)]);
```
Comment (mhevery) [inline app/app.js:7]: Use fast changed detection

File: app/components/calendar-cell/calendar-cell.js [hunk 2]
```diff
@@ -19,8 +20,12 @@ var randomMillis = function() {
   ]
 })
 export class CalendarCell {
+	bpc;
 
-  constructor() {
+  constructor(bpc:BindingPropagationConfig) {
+    this.isPure = true;
```
Comment (mhevery) [inline app/components/calendar-cell/calendar-cell.js:26]: Make component pure

File: app/components/calendar-cell/calendar-cell.js [hunk 3]
```diff
@@ -39,7 +44,8 @@ export class CalendarCell {
       setTimeout(() => {
         self.status.isSearching = false;
         self.status.searchResults.options = Math.floor(Math.random() * 5);
-      }, randomMillis());
+        if (this.isPure) this.bpc.shouldBePropagated();
+      }, 0);
```
Comment (mhevery) [inline app/components/calendar-cell/calendar-cell.js:48]: We are faster than 500ms so to make the UI go faster do instant.

**Comment by djsmith42** [comment]
@mhevery This is amazing. I can't believe how fast this is. I am just totally blown away. Thanks very much!

The future of Angular2 is incredibly bright.

**Pull request merged**